app/models/login_user.rb

```diff
@@ -7,7 +7,7 @@ def perform
   private
 
   def user
-    @_user ||= User.find_or_create_by(github_id: auth_hash[:uid])
+    @_user ||= FindOrRegisterUser.new(github_id: auth_hash[:uid]).perform
```

Comment:
in this class we are also calling `from_oauth_hash(auth_hash)`, which sets attributes and creates the user. maybe move that logic into this new class?